bpf/lib/nat.h

```diff
@@ -395,23 +395,17 @@ snat_v4_icmp_rewrite_ingress_embedded(struct __ctx_buff *ctx,
 			 * original.
 			 */
 			if (ctx_store_bytes(ctx, inner_l4_off +
-					    offsetof(struct icmphdr, un.echo.id),
-					    &state->to_dport,
-					    sizeof(state->to_dport), 0) < 0)
+						offsetof(struct icmphdr, un.echo.id),
+						&state->to_dport,
+						sizeof(state->to_dport), 0) < 0)
 				return DROP_WRITE_ERROR;
-			csum.offset = offsetof(struct icmphdr, checksum);
-			csum.flags = 0;
+			if (l4_csum_replace(ctx, inner_l4_off + offsetof(struct icmphdr, checksum),
+					    tuple->dport,
+					    state->to_dport,
+					    sizeof(tuple->dport)) < 0)
+				return DROP_CSUM_L4;
 			break;
-		}
-		default:
-			return DROP_UNKNOWN_L4;
-		}
-		/* By recomputing L4 checksum of inner packet we avoid having
-		 * to recompute L4 of the ICMP Error.
-		 */
-		suml4 = csum_diff(&tuple->dport, 4, &state->to_dport, 4, 0);
-		if (csum_l4_replace(ctx, inner_l4_off, &csum, 0, suml4, 0) < 0)
```

Comment:
I understand the bug you want to fix: the original `csum_l4_replace` wasn't called with flag `BPF_F_PSEUDO_HDR`. Then how about simply passing a `flag` variable to it?